lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -178,7 +170,6 @@ export abstract class ClientApplication {
                 this.logger.verbose(e);
             }
         }
-        this.initialized = true;
```

Comment:
Maybe we should think about this in a different way.  We can leave the `protected initialized: boolean;` in place on line 84 above and keep the `this.initialized = true` here when the initialize call is made.  We can then check if initilized != true and fail library calls.  I'm not sure what the behavior is if someone fails to call initialize and calls our other APIs.  We might be able to clarify it today but as we go further along, we might find that some APIs that were once working start to fail in weird ways in cases where Initialize was not called.  Thoughts?

Reply:
That approach is what I initially had in mind as well but I actually don't mind the approach in this PR, the developer experience feels a bit cleaner/more intuitive to me. I do think it's a good idea to throw if initialize hasn't been called though regardless of which approach we take.

Reply:
I am fine with keeping `initliazed` variable though i think it would be redundant since we enforce to instantiate an application with a static method that always invokes `initliaze()` method.

Reply:
Unfortunately `protected` is a TypeScript construct and will be removed at transpile time so we can't prevent people from calling the PCA constructor directly. We'll of course say that's unsupported but throwing an error here is still a good idea IMO